src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,69 @@
+package duke.parser;
+
+import duke.exception.DukeException;
+
+/**
+ * A parser for reading user input
+ */
+public class Parser {
+    private String msg;
+
+    /**
+     * Constructor for <code>Parser</code>.
+     *
+     * @param msg a string of input from user
+     */
+    public Parser(String msg) {
+        this.msg = msg;
+    }
+
+    /**
+     * Process msg and return the request (command) of the input.
+     *
+     * @return a string of request
+     */
+    public String getRequest() {
+        String[] tk = msg.split(" ");
+        return tk[0];
```

Comment:
Perhaps use a more descriptive name other than `tk`? 
I noticed several other places using this variable name as well.